docs/contributing/design-documents.rst

```diff
@@ -0,0 +1,74 @@
+:title: Design Documents
+:description: Information necessary for a feature design document.
+
+.. _design-documents:
+
+Design Documents
+================
+
+Before submitting a pull request which is expected to significantly change the
+behavior of Deis (such as when contributing a new feature or refactor),
+contributors should first open an issue representing a design document.
+
+Goals
+-----
+
+Design documents were adopted in order to:
+
+* Involve stakeholders as early as possible in a feature's development
+* Ensure code changes accomplish the original motivations and design goals
+* Establish clear acceptance criteria for a feature or change
+* Enforce test-driven design methodology and automated test coverage
```

Comment:
Excellent point. The Deis project is going to stop making any exceptions for PRs without tests (which may require us to beef up our test coverage and create test harnesses in some places).